File: backend/business-service/src/main/java/com/txakurrapp/businessservice/model/BusinessOwner.java
package com.txakurrapp.businessservice.model;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
public class BusinessOwner {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private LocalDate signUpDate;
    private Long userId;
    private String firstName;
    private String lastName;
    private String personalId;
    private LocalDate birthDate;
    private Integer phoneNumber;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "address_id")
    private Address address;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "image_id")
    private Image profilePicture;

    @OneToMany(mappedBy = "businessOwner", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Service> services;

    public BusinessOwner() {
    }

    public BusinessOwner(Long userId, String firstName, String lastName, String personalId, LocalDate birthDate, Integer phoneNumber, Address address, Image profilePicture) {
        this.signUpDate = LocalDate.now();
        this.userId = userId;
        this.firstName = firstName;
        this.lastName = lastName;
        this.personalId = personalId;
        this.birthDate = birthDate;
        this.phoneNumber = phoneNumber;
        this.address = address;
        this.profilePicture = profilePicture;
    }


//  Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getSignUpDate() {
        return signUpDate;
    }

    public void setSignUpDate(LocalDate signUpDate) {
        this.signUpDate = signUpDate;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getPersonalId() {
        return personalId;
    }

    public void setPersonalId(String personalId) {
        this.personalId = personalId;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate = birthDate;
    }

    public Integer getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(Integer phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public Image getProfilePicture() {
        return profilePicture;
    }

    public void setProfilePicture(Image profilePicture) {
        this.profilePicture = profilePicture;
    }

    public List<Service> getServices() {
        return services;
    }

    public void setServices(List<Service> services) {
        this.services = services;
    }
}

File: frontend/txakurrapp/src/app/classes/business/business-owner.ts
import { Image } from "./image";
import { Address } from "./address";
import { PetService } from "./pet-service";

export class BusinessOwner {

    constructor(
        private _id: number,
        private _signUpDate: Date,
        private _userId: number,
        private _firstName: string,
        private _lastName: string,
        private _personalId: string,
        private _birthDate: Date,
        private _phoneNumber: number,
        private _address: Address,
        private _profilePicture: Image,
        private _services: PetService[]
    ){}


    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get signUpDate(): Date {
        return this._signUpDate;
    }
    public set signUpDate(value: Date) {
        this._signUpDate = value;
    }
    public get userId(): number {
        return this._userId;
    }
    public set userId(value: number) {
        this._userId = value;
    }
    public get firstName(): string {
        return this._firstName;
    }
    public set firstName(value: string) {
        this._firstName = value;
    }
    public get lastName(): string {
        return this._lastName;
    }
    public set lastName(value: string) {
        this._lastName = value;
    }
    public get personalId(): string {
        return this._personalId;
    }
    public set personalId(value: string) {
        this._personalId = value;
    }
    public get birthDate(): Date {
        return this._birthDate;
    }
    public set birthDate(value: Date) {
        this._birthDate = value;
    }
    public get phoneNumber(): number {
        return this._phoneNumber;
    }
    public set phoneNumber(value: number) {
        this._phoneNumber = value;
    }
    public get address(): Address {
        return this._address;
    }
    public set address(value: Address) {
        this._address = value;
    }
    public get profilePicture(): Image {
        return this._profilePicture;
    }
    public set profilePicture(value: Image) {
        this._profilePicture = value;
    }
    public get services(): PetService[] {
        return this._services;
    }
    public set services(value: PetService[]) {
        this._services = value;
    }

}


